site-content/source/modules/ROOT/pages/blog/Configurable-Storage-Ports-and-Why-We-Need-Them.adoc

```diff
@@ -0,0 +1,36 @@
+= Configurable Storage Ports and Why We Need Them
+:page-layout: single-post
+:page-role: blog-post
+:page-post-date: December 28, 2021
+:page-post-author: Chris Thornett
+:description: The Apache Cassandra Community
+:keywords: 
+
+= Configurable Storage Ports and Why We Need Them
+
+image::blog/configurable-storage-ports-and-why-we-need-them-unsplash-bernard-hermant.jpg[storage ports]
+
+https://unsplash.com/@bernardhermant?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText[Bernard Hermant on Unsplash,window=_blank]
+
+As a distributed database, Cassandra relies heavily on running across multiple machines and communicating over a network. Since we take pride in keeping Cassandra lightweight, it is important that network configuration is highly adaptable. This allows Cassandra nodes to communicate over a greater variety of networks and network devices, across firewalls, and between private networks.
+Cassandra nodes talk to each other just like most internet services: over TCP/IP connections. The storage port is the TCP port number used specifically for inter-node communication. By default, this is set to port 7000—check out the `storage_port` setting in the stock *cassandra.yaml* file.
+
+=== Why Would I Need to Change the storage_port Setting?
+There are a few instances where the storage_port setting might commonly need to be changed. You might be trying to run multiple Cassandra instances or configurations on one server, and that server could only have one IP address. In this case, each Cassandra node will need to use its own `storage_port`.
+Network configurations can also require different `storage_port` settings. For example, firewall restrictions, port forwarding at upstream routers, and even network address translation issues all have the possibility of requiring different port settings.
+Finally, there may be a conflict on port 7000. Perhaps some other application or service needs to use that port.
+
+=== How Do My Other Cassandra Nodes Know About Different storage_port Settings?
+If you need to change the `storage_port` setting for one of your Cassandra nodes, you will probably have to change the `storage_port` setting for all of the nodes in that ring. If you have multiple rings across different data centers, then you will need to set up some kind of port-forwarding solution at the firewall or router level if those data centers use different `storage_port` settings.
```

Comment:
Nitpick: replace 'you will probably have to' with ' you will need to' in the first sentence:

> ... your Cassandra nodes, you will probably have to change the `storage_port` setting ...